miqa/core/tests/test_rest.py

```diff
@@ -62,6 +62,34 @@ def test_projects_list(user_api_client, project, user):
         }
 
 
+@pytest.mark.django_db
+def test_project_status(
+    user_api_client,
+    project,
+    experiment,
+    scan_factory,
+    scan_decision_factory,
+    user_factory,
+):
+    # 3 of 5 scans marked as complete
+    decisions = [
+        ('U', 'tier_1_reviewer'),
+        ('U', 'tier_2_reviewer'),
+        ('UN', 'tier_1_reviewer'),  # needs tier 2 review
+        ('UN', 'tier_2_reviewer'),
+        ('Q?', 'tier_1_reviewer'),  # needs tier 2 review
+    ]
+    scans = [scan_factory(experiment=experiment) for i in range(len(decisions))]
+    for i, scan in enumerate(scans):
+        decider = user_factory()
+        assign_perm(decisions[i][1], decider, project)
+        scan_decision_factory(scan=scan, creator=decider, decision=decisions[i][0])
+    resp = user_api_client().get(f'/api/v1/projects/{project.id}')
+    if resp.status_code == 200:
```

Comment:
It's usually not good to have conditional blocks inside the test body. In this case, wouldn't we just want to assert that the status code was 200, rather than silently passing if the request fails?

Reply:
I agree. I had put this there because our `user_api_client` fixture is parameterized, so the test will run once for a superuser-authenticated client, once for a tier_2_reviewer-authenticated, once for collaborator-authenticated, and once for no project access. In the last case, the project endpoint returns a `404`.
But, with the other change you suggested, we don't need to check the status code because we don't access the endpoint at all. And without using the `user_api_client` fixture, the test only runs once :)